packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +67,693 @@ export class ModelMesh extends Mesh {
+
+    if (uv.length !== this._vertexCount) {
+      throw "The array provided needs to be the same size as vertex count.";
+    }
+
+    channelIndex = channelIndex ?? 0;
+    switch (channelIndex) {
+      case 0:
+        this._vertexSlotChanged = !!this._uv !== !!uv;
+        this._vertexChangeFlag |= ValueChanged.UV;
+        this._uv = uv;
+        break;
+      case 1:
+        this._vertexSlotChanged = !!this._uv1 !== !!uv;
+        this._vertexChangeFlag |= ValueChanged.UV1;
+        this._uv1 = uv;
+        break;
+      case 2:
+        this._vertexSlotChanged = !!this._uv2 !== !!uv;
+        this._vertexChangeFlag |= ValueChanged.UV2;
+        this._uv2 = uv;
+        break;
+      case 3:
+        this._vertexSlotChanged = !!this._uv3 !== !!uv;
+        this._vertexChangeFlag |= ValueChanged.UV3;
+        this._uv3 = uv;
+        break;
+      case 4:
+        this._vertexSlotChanged = !!this._uv4 !== !!uv;
+        this._vertexChangeFlag |= ValueChanged.UV4;
+        this._uv4 = uv;
+        break;
+      case 5:
+        this._vertexSlotChanged = !!this._uv5 !== !!uv;
+        this._vertexChangeFlag |= ValueChanged.UV5;
+        this._uv5 = uv;
+        break;
+      case 6:
+        this._vertexSlotChanged = !!this._uv6 !== !!uv;
+        this._vertexChangeFlag |= ValueChanged.UV6;
+        this._uv6 = uv;
+        break;
+      case 7:
+        this._vertexSlotChanged = !!this._uv7 !== !!uv;
+        this._vertexChangeFlag |= ValueChanged.UV7;
+        this._uv7 = uv;
+        break;
+      default:
+        throw "The index of channel needs to be in range [0 - 7].";
+    }
+  }
+
+  /**
+   * Get uv for the mesh.
+   * @remarks Please call the setUV() method after modification to ensure that the modification takes effect.
+   */
+  getUVs(): Vector2[] | null;
+  /**
+   * Get uv for the mesh by channelIndex.
+   * @param channelIndex - The index of uv channels, in [0 ~ 7] range.
+   * @remarks Please call the setUV() method after modification to ensure that the modification takes effect.
+   */
+  getUVs(channelIndex: number): Vector2[] | null;
+  getUVs(channelIndex?: number): Vector2[] | null {
+    if (!this._accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+    channelIndex = channelIndex ?? 0;
+    switch (channelIndex) {
+      case 0:
+        return this._uv;
+      case 1:
+        return this._uv1;
+      case 2:
+        return this._uv2;
+      case 3:
+        return this._uv3;
+      case 4:
+        return this._uv4;
+      case 5:
+        return this._uv5;
+      case 6:
+        return this._uv6;
+      case 7:
+        return this._uv7;
+    }
+    throw "The index of channel needs to be in range [0 - 7].";
+  }
+
+  private _updateVertexElements(): VertexElement[] {
+    const vertexElements = [POSITION_VERTEX_ELEMENT];
+    const { _colors, _normals, _tangents, _uv, _uv1, _uv2, _uv3, _uv4, _uv5, _uv6, _uv7, _boneWeights: _weights, _boneIndices: _joints } = this;
```

Comment:
Only  use once, can use this.xx directly.

Reply:
+1